File: service/Api/Subscription/Subscription.cs
using System;

namespace Quizmeister
{
    public class Subscription
    {
        public int SubscriptionId { get; set; }
        public string ShortId { get; set; }
        public SubscriptionStatus Status { get; set; }
        public string Team { get; set; }
        public string Email { get; set; }
        public string Remark { get; set; }
        public DateTime ModifiedOn { get; set; }
        public string SearchField { get; set; }
        public Quiz Quiz { get; set; }

    }
}

File: app/projects/domain/src/lib/subscriptions/subscription.ts
import { MessageStatus } from '@domain/infrastructure/message-status';
import { Quiz } from '../quiz/quiz';

export interface NewSubscriptionEvent {
  subscription: Subscription;
  status: MessageStatus;
  message: string;
}

export interface UpdateSubscriptionEvent {
  subscription: Subscription;
  status: MessageStatus;
  message: string;
}

export interface DeleteSubscriptionEvent {
  subscription: Subscription;
  status: MessageStatus;
  message: string;
}

export interface Subscription {
  default: boolean;
  subscriptionId: number;
  quiz: Quiz;
  status: SubscriptionStatus;
  email: string;
  team: string;

  answered: boolean;
}

export class SubscriptionDefault implements Subscription {
  default = true;
  subscriptionId: number;
  quiz: Quiz;
  status: SubscriptionStatus;
  email: string;
  team: string;
  answered: boolean;
}

/* eslint-disable no-shadow */
export enum SubscriptionStatus {
  new = 1,
  queued = 4,
  joined = 2,
  finished = 3,
}
